### PR TITLE
Refactors `BlockRequest` and adds tests for it

### PR DESCRIPTION
**Pull request by howardwu**
## Motivation

Refactors `BlockRequest` and adds tests for it

## Test Plan

Adds the following tests:
- `test_block_request`
- `test_block_request_eq`

### CODE REVIEW

File: src/helpers/block_request.rs [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        }
+    }
+
+    #[test]
+    fn test_block_request_eq() {
+        let rng = &mut thread_rng();
+
+        let a = BlockRequest::<Testnet2>::from(1000);
+        let b = BlockRequest::<Testnet2>::from((1000, None));
+        assert_eq!(a, b);
```
Comment (niklaslong) [inline src/helpers/block_request.rs]: Why no iteration in this case?

File: src/helpers/block_request.rs [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use snarkvm::{dpc::testnet2::Testnet2, prelude::UniformRand};
+
+    use rand::{thread_rng, Rng};
+
+    const ITERATIONS: usize = 1_000_000;
```
Comment (niklaslong) [inline src/helpers/block_request.rs]: I don't see any harm in iterating (beyond making the tests slower) but is there any reason you suspect different values would change the behaviour in this case?

Comment (ljedrz) [reply src/helpers/block_request.rs]: For this many iterations I'd rather use `thread_rng` only to seed a different, faster source of randomness that would be used in the loop; it's something I'd like to do for other tests too, including snarkVM ones.

Comment (howardwu) [reply src/helpers/block_request.rs]: We can do that in one go to keep it consistent in design.

The rationale for multiple iterations is more to ensure the Default::default() block hash Eq case is accounted for.

I put 1 million iterations arbitrarily, it could be 1000 as well.

File: src/helpers/block_request.rs (hunk 1 above)
Comment (howardwu) [reply src/helpers/block_request.rs]: Good point, will add

File: src/helpers/block_request.rs [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+    #[test]
+    fn test_block_request() {
+        let rng = &mut thread_rng();
+
+        for _ in 0..ITERATIONS {
```
Comment (ljedrz) [inline src/helpers/block_request.rs]: I'm not sure if running these checks for many random heights and hashes is more likely to detect issues than just testing 2 cases (any height + `None` and any height + `Some(any hash)`) for equality and conversion; after all, the actual values of the hash and height are not converted further or used in any operations.

**Review by ljedrz — COMMENTED** [review]
Moving the object to its own module is a good idea, but I'm not convinced that running many iterations for random values is more likely to detect issues, as commented [here](https://github.com/AleoHQ/snarkOS/pull/1529/files#r780128641).

File: src/helpers/block_request.rs (hunk 1 above)
Comment (howardwu) [reply src/helpers/block_request.rs]: Added it in

File: src/helpers/block_request.rs (hunk 2 above)
Comment (howardwu) [reply src/helpers/block_request.rs]: I dropped the iteration count to 5.

File: src/helpers/block_request.rs (hunk 3 above)
Comment (howardwu) [reply src/helpers/block_request.rs]: I dropped the iteration count to 5.

**Review by raychu86 — APPROVED** [review]
